lib/uplink/project.go

```diff
@@ -157,6 +157,20 @@ func (p *Project) OpenBucket(ctx context.Context, bucketName string, access *Enc
 		return nil, err
 	}
 
```

Comment:
**Rational for code Changes:** 
This function is updated to handle the addition of 'PartnerID' at the time of bucket creation or at a later time after the bucket was first created and no partnerID was associated with it when it was first created. 

see the above test case **"open with partner id"** and **"open with different partner id"** respectively